### ISSUE TITLE
Ignoring lint rules in templates

### ISSUE THREAD

**Issue by MaxMilton** [sveltejs/eslint-plugin-svelte3#22]
Now that `v2.0.0` has template linting is on by default, I'm getting a bunch of false negatives I'd like to either ignore or customise specifically for templates.

The main 2 I'm seeing (based on my rule set) is:

1. `no-unused-expressions` - We kind of actually want the opposite, templates _should be_ an expression.
1. `@typescript-eslint/indent` (same as `indent`) - Triggers inside an `{#each}` loop.

![Screenshot from 2019-06-26 06-54-33](https://user-images.githubusercontent.com/14946546/60132871-b879c500-97df-11e9-8e1c-5f99396b98cb.png)

Configs:

- https://github.com/WeAreGenki/minna-ui/blob/dev/utils/eslint-config/index.js
- https://github.com/WeAreGenki/minna-ui/blob/dev/utils/eslint-config/svelte.js

**Comment by Conduitry** [comment]
Having this plugin explicitly disable the `no-unused-expressions` rule in template expressions makes sense.

I'm not sure what to do about the `@typescript/` rule. I'd really like to be able to draw a line about not having any special treatment for rules from any plugin (because where would the feature requests for this stop?). For this particular case, it's probably safe to use your `*.svelte` override to completely disable `@typescript-eslint/indent` in Svelte components.

Adding back an option to disable template linting entirely would probably make things smoother but it doesn't feel like the best move. I'm also worried about adding a general 'do you want to disable this message' callback in the settings, that's passed a `message` object from ESLint (and maybe also whether this corresponds to the template or a script) and returns a boolean. I guess overall I'd really like to push people toward configuring ESLint how they need it, rather than configuring this plugin how they need it. `@typescript-eslint/indent` is an example of this - there's already an official mechanism provided by ESLint for disabling this on specific files.

**Comment by Conduitry** [comment]
`no-unused-expressions` is now getting ignored in expressions in the template.

**Comment by MaxMilton** [comment]
I agree with you about limiting fixes to built-in ESLint rules only. Easier to maintain and keeps the scope of integration testing tight.

Yeah, adding an option to disable template linting feels like a step in the wrong direction. Svelte doesn't have first-class TypeScript support yet so `@typescript-eslint/indent` could be replaced with regular `indent` for now. Even still, adding overrides for `*.svelte` when really we just want to target svelte templates isn't really desirable either.

The [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc#settings-to-configure-check-examples) has a similar problem and have a solution; They have a `settings.jsdoc.matchingFileName` setting which allows the user to change the file name passed to ESLint for matching code blocks. This then allows setting up overrides for those specific code blocks. Using the overrides mechanism allows for full config flexibility.

Not sure how viable it would be for this plugin but I can imagine setting svelte template expressions to pass a `component.svelte-template` file name and adding a new set of overrides.

---

For anyone running into the same original issue, you don't need to disable `@typescript-eslint/indent` completely, just ignore the `Identifier` node like so:

```js
'@typescript-eslint/indent': ['error', 2, {
    ignoredNodes: ['Identifier'], // Svelte template expressions
  }
]
```

You could probably use a more specific AST pattern but this is better than nothing.

**Comment by Conduitry** [comment]
[Processors in ESLint 6 got a bit of a overhaul](https://eslint.org/docs/user-guide/configuring#specifying-processor), including a new feature called named code blocks, which seem like now the correct way to expose the ability to have different configurations for different sections of a Svelte component. I have [a pending question](https://gitter.im/eslint/eslint?at=5d267e510c8daa1686c113ef) about how these work, but in the meantime I have an implementation that uses these in [my fork of this plugin](https://github.com/Conduitry/eslint-plugin-svelte3).

**Comment by Conduitry** [comment]
This is implemented in 2.7.0 via [named code blocks](https://github.com/sveltejs/eslint-plugin-svelte3#named-code-blocks).

**Comment by Conduitry** [comment]
Adding named code blocks was actually a breaking change 😬 I've released 2.7.1 where they're disabled by default, but they can be enabled with `svelte3/named-blocks`.

**Comment by MaxMilton** [comment]
It works. We can now create presise overrides :tada:

So useful, thank you!
